src/rust-2024/rustdoc-doctests.md

```diff
@@ -0,0 +1,76 @@
+/// ```
+pub fn subtract(left: u64, right: u64) -> u64 {
+    left - right
+}
+```
+
+In this example, the two doctests will now be compiled in a single executable. Rustdoc will essentially place each example in a separate function within a single binary. The tests still run in independent processes as they did before, so any global state (like global statics) should still continue to work correctly.
+
+This change is only available in the 2024 Edition to avoid potential incompatibilities with existing doctests which may not work in a combined executable.
+
+[doctests]: ../../rustdoc/write-documentation/documentation-tests.html
+[libtest harness]: ../../rustc/tests/index.html
+
+### Standalone tag
+
+In some situations it is not possible for rustdoc to combine examples in a single executable. Rustdoc will attempt to automatically detect if this is not possible, for example:
+
+* Uses the [`compile_fail`][tags] language tag, which indicates that the example fails to compile.
+* Uses the [`edition`][tags], which indicates the edition of the example.[^edition-tag]
+* Uses global attributes like the [`global_allocator`] attribute, which could potentially interfere with other tests.
+* Defines any crate-wide attributes (like `#![deny(unused)]`).
+* Defines a macro that uses `$crate`, because the `$crate` path will not work correctly.
+
+However, rustdoc is not able to automatically determine *all* situations where an example cannot be combined with other examples. In these situations, you can add the `standalone` language tag to indicate that the example should be built as a separate executable.
+
+```rust
+//! ```
+//! let location = std::panic::Location::caller();
+//! assert_eq!(location.line(), 5);
+//! ```
+```
+
+This example is sensitive to the code structure of how the example is compiled. This won't work with the "combined" approach because the line numbers will shift depending on how the doctests are combined. In these situations, you can add the `standalone` tag to force the example to be built separately just as it did in previous editions.
+
+```rust
+//! ```standalone
+//! let location = std::panic::Location::caller();
+//! assert_eq!(location.line(), 5);
+//! ```
+```
+
+[tags]: ../../rustdoc/write-documentation/documentation-tests.html#attributes
+[`global_allocator`]: ../../std/alloc/trait.GlobalAlloc.html
+
+[^edition-tag]: Note that rustdoc will only combine tests if the entire crate is Edition 2024 or greater. Using `edition2024` tags in older editions will not result in those tests being combined.
+
+## Migration
+
+There is no automatic migration to determine which doctests need to be annotated with the `standalone` tag. You will need to update your crate to the 2024 Edition and then run your documentation tests and see if any of them fail. If they do fail, you will need to analyze whether the test can be rewritten to be compatible with the combined approach, or add the `standalone` tag to retain the previous behavior.
```

Comment:
I would "tone it down" a bit. In all crates we tested, we only found one case (in `libcore`) which didn't work with merged doctests. I think maybe something "It's very unlikely to happen but some doctests might not worked when merged. The only way to know it is to run them with the 2024 edition. ...". What do you think?

Reply:
Yea, I thought about adding something along those lines. I added a few sentences to emphasize that it is unlikely to be an issue.